### ISSUE TITLE
Mini player becomes immovable if placed under the Mac OS toolbar

### ISSUE THREAD

**Issue by SoftVision-EmilPasca** [meandavejustice/min-vid#499]
[Affected versions]:
- Firefox 45.0 and up
- Min-Vid 0.3.3-dev

[Affected Platforms]:
- All Mac OS

[Prerequisites]:
- Latest Min-Vid add-on version (0.3.3-dev) was previously installed on a clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. Navigate to www.youtube.com page.
3. Send a video to the mini player using the thumbnail overlay or context menu.
4. Drag the mini player until the top control buttons are behind the OS toolbar.
5. Release the mouse button and after try to drag the mini player to a new position.

[Expected result]:
- Mini player is again grabbed and can be moved to a new position.

[Actual result]:
- Mini player remains stuck under the OS toolbar.

[Additional Notes]:
- The mini player cannot be moved, or closed. The only way to close it is by fast forwarding to the end of video playback and then click the "X" button or by restarting the browser.
- Attached a screen recording of the issue:
![issue_minvid_macos](https://cloud.githubusercontent.com/assets/20083658/20394234/8aeb158e-ace7-11e6-995f-956ec62bc9f1.gif)

**Comment by meandavejustice** [comment]
Is this still an issue after the changes in #475 ?

**Comment by SoftVision-PaulOiegas** [comment]
@meandavejustice Yes, this still occurs with the build made today (12/23/2016) from the repo.

**Comment by SoftVision-CiprianMuresan** [comment]
I have found a different scenario on Windows only, it seems to have the same root cause as the scenario above.

[Affected versions]:
- Firefox 51.0 and up
- Min-Vid 0.3.4-dev (built on 2/9/2017)

[Affected Platforms]:
- All Windows

[Prerequisites]:
- Latest Min-Vid add-on version (0.3.4-dev) was previously installed on a clean profile.

[Steps to reproduce]:
1. Open the browser with the profile from prerequisites.
2. Navigate to www.youtube.com page.
3. Send a video to the mini player and change the player position to the bottom of the screen (a bit over the taskbar).
4. Minimize the player and close it.
5. Send a video to the mini player and observe the player position.
(bonus round) 6. Minimize the player.

[Expected result]:
Step 5. The player is positioned in such a way that it is fully visible on the screen.
(bonus round) Step 6. The player is positioned in such a way that it is fully visible on the screen. 

[Actual result]:
Step 5. Half the player is under the taskbar.
(bonus round) Step 6. The player is no longer visible on the screen.

[Notes]:
- After doing this, the mini player can no longer be brought back on the screen unless you restart the browser or disable and re-enable the add-on.
- Attached a screen recording of the issue:
![mini player position](https://cloud.githubusercontent.com/assets/18304511/22790701/bb83e974-eeef-11e6-8dc5-975f8c7ef963.gif)

**Comment by SoftVision-CiprianMuresan** [comment]
Updating the title of this issue, because it is also reproducible on Windows.
